### PR TITLE
Update gitian setup

### PR DESCRIPTION
**Pull request by charlieok**
This PR asks users of zcash-gitian to adopt a slightly different workflow -- setting environment variables rather than editing the version-tracked gitian.yml.

Changes:
- Update to current versions of python and ansible
- Walk through (optional) python virtual environment setup local to the project for ansible install
- Use environment variables for inputs
- Use local .gitignored files for user customizations rather than asking user to edit a version-tracked file
- Read from user's git config for name/email rather than asking user to enter that same info in another place
- Walk through setting up direnv as suggested tool for automatically setting environment variables
- Change 'gpg2' commands to 'gpg' and remove steps to set up shims that were needed to make gpg2 work.
- Add more explanation on how the gpg_key_name input is used (answer: to set gitian's "signer" input)

Updated for and tested on current versions of macOS, Debian, and Ubuntu.

Built and pushed signatures from this branch on Ubuntu 18.04:
https://github.com/zcash/gitian.sigs/pull/32

### CODE REVIEW

**Comment by garethtdavies** [comment]
I haven't tested this yet but plan to soon on a fresh machine but one of the minor issues for me is that every time I have to run a `git remote set-url origin my-gitian-sigs-repo`. Is there any possibility of adding the Gitian sigs repo as an environment variable too as I am sure this would make life much easier for those who don't have commit access to the Zcash one. i.e.

`ZCASH_GITIAN_SIGS_REPO_URL=https://github.com/zcash/gitian.sigs`

I also think a couple of lines of how to create a new branch to submit a PR would also help, i.e.

```
git checkout -b v2.0.4
git push origin v2.0.4
```

**Comment by charlieok** [comment]
Ran to completion when testing `v2.0.4` on macOS and looks like I got matching hashes to previously posted sigs

```
vagrant@zcash-build:~/gitian.sigs/v2.0.4$ head charlie/zcash-2.0.4-build.assert
--- !!omap
- out_manifest: |
    052e7de863e1e68f7d0886f5dbd421fa6109688f91859c0dee0a09e94f0d6e6f  src/zcash-2.0.4.tar.gz
    f0f8c46ef92d05fdb6d96c97fbc04c702f95c0adef571b88ea253bac754162b3  zcash-2.0.4-linux64-debug.tar.gz
    a825060f425128260b603e3524c472f4ba28ba17090bece4ac38c84b1bcf5940  zcash-2.0.4-linux64.tar.gz
- in_manifest: |-
    5d3ff6f7f3976b52e07aac84fda95ef827c0dc8372fca8f217bd8bab6a5a1b39  zcash-2.0.4-desc.yml
    git:be1d68ef763ce405d4d04d7f4d3dfbbdd9084687 zcash
- base_manifests: !!omap
  - jessie-amd64: |
```

**Comment by charlieok** [comment]
Thanks for looking at this @garethtdavies ! I hadn't even posted it in chat yet, was still testing :)

I 100% agree with the pain points you mentioned, and have been meaning to pay some more attention there. This PR is basically an accumulation of things I managed to get to when release/signature time rolled back around.

**Comment by charlieok** [comment]
@garethtdavies I added a bit of your suggestion about describing more "how to push signatures". I like the `ZCASH_GITIAN_SIGS_REPO_URL` suggestion too but I'll have to leave that one for later.

File: dependency_install_steps_by_platform/Ubuntu_18.04.x.md [hunk 1]
```diff
@@ -2,46 +2,61 @@
...
+```
 
-## Install Vagrant 2.0.3 or higher
 
-As of this writing, the vagrant version that Ubuntu 18.04 uses is 2.0.2 so we suggest a later
-release in the 2.0.x series.
+
+## Choice: Install Ansible via apt now or pip later
+
+You can install apt to a system-wide location using Ubuntu's apt tool, which will be a less current
```
Comment (garethtdavies) [inline dependency_install_steps_by_platform/Ubuntu_18.04.x.md]: Small typo here: "You can install Ansible to a system..."

File: README.md [hunk 2]
```diff
@@ -247,13 +473,13 @@ This way, we don't need to copy ssh keys into the VM. You can verify that the ke
...
+zcash-gitian$ ssh-add -l
 The agent has no identities.
 
-$ ssh-add ~/.ssh/github_id_rsa
+zcash-gitian$ ssh-add ~/.ssh/github_id_rsa
 Identity added: /home/hpotter/.ssh/github_id_rsa (/home/hpotter/.ssh/github_id_rsa)
 
-$ ssh-add -l
+zcash-gitian$ ssh-add -l
 4096 SHA256:4fFdwJ71VIpF5cW0dqrsU7jxjctaFcAKmdQZPEqR0Y4 /home/hpotter/.ssh/github_id_rsa (RSA)
```
Comment (garethtdavies) [inline README.md]: Minor but the command to generate the ssh key would produce a 2048 bit key I think by default. Should it be updated to generate a 4096 bit key as per this one `ssh-keygen -t rsa -b 4096 -C "hpotter@hogwarts.wiz" -f ~/.ssh/github_id_rsa -N ''`

**Comment by garethtdavies** [comment]
Followed this on a new build machine. I used existing SSH and GPG keys that I imported but other than that followed exactly. I built on the latest Disco Dingo Ubuntu 19.04 (development). 

The only bit that briefly stumped me was the `direnv` stuff as I had to search for how to install this. Probably adding a line in the installing dependencies would cure this i.e. Ubuntu `apt get install direnv` and then a line that it needs to be hooked into each shell (I'd probably add the example for BASH and then link to docs for other shells). 

Pushed the new sigs here: https://github.com/garethtdavies/gitian.sigs/blob/3161a5effec995f27fe07106e8501299703ec57a/v2.0.4/hello/zcash-2.0.4-build.assert and they match! :+1:

File: dependency_install_steps_by_platform/Ubuntu_18.04.x.md (hunk 1 above)
Comment (charlieok) [reply dependency_install_steps_by_platform/Ubuntu_18.04.x.md]: good catch, thanks!

File: README.md (hunk 2 above)
Comment (charlieok) [reply README.md]: seems reasonable; assuming this is meant for the line with the `ssh-keygen` command

**Comment by charlieok** [comment]
@garethtdavies oops, I actually did have sections on `direnv` in the macOS and Debian pages but forgot to add it when updating the Ubuntu page. Fixed.

Thanks again for the close read, test, and signature!

**Pull request merged**